core/src/main/scala/org/broadinstitute/dsde/rawls/entities/EntityService.scala

```diff
@@ -124,6 +124,25 @@ class EntityService(protected val userInfo: UserInfo, val dataSource: SlickDataS
       }
     }
 
+  def renameEntityType(workspaceName: WorkspaceName, oldName: String, newName: String): Future[Int] =
+    getWorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.write, Some(WorkspaceAttributeSpecs(all = false))) flatMap { workspaceContext =>
+      dataSource.inTransaction { dataAccess =>
+        dataAccess.entityQuery.doesEntityTypeAlreadyExist(workspaceContext, newName) flatMap {
+          _.head match {
+            case false =>
+              dataAccess.entityQuery.doesEntityTypeAlreadyExist(workspaceContext, oldName) flatMap {
+                _.head match {
+                  case false => throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.NotFound,
+                    s"Can't find entity type ${oldName}"))
+                  case true => dataAccess.entityQuery.changeEntityTypeName(workspaceContext, oldName, newName)
+                }
+              }
+            case true => throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Conflict, s"${newName} already exists as an entity type"))
+          }
+        }
+      }
```

Comment:
This is admittedly a mess.  I've tried simplifying to `if(_.head)..` but the compiler doesn't recognize the variable in that case.  I also originally had my `workspaceQuery.updateLastModified(workspaceContext.workspaceIdAsUUID)` call immediately following the db call to change the entity type and that to a 204 response every time but the db never updating.  Any tips, enlightenment especially appreciated here.

Reply:
to get the `if` clauses working, grab the result of the `flatMap` and use that result instead of using `_`:
```scala
        dataAccess.entityQuery.doesEntityTypeAlreadyExist(workspaceContext, newName) flatMap { newNameExists =>
          if (newNameExists.head)
```
but also note that `head` will throw `NoSuchElementException` if `newNameExists` is empty